src/components/CalendarGroups/CalendarGroups.js

```diff
@@ -82,12 +84,27 @@ const CalendarGroups = ({
               date={selectedDate}
               key={`groupColumn${index}`}
               columnKey={`groupColumn${index}`}
-              columnIndex={index}
+              columnIndex={columnIndex}
               columnId={eventGroup.id}
               eventsForColumn={eventsForDay}
               stepDetailsForColumn={stepDetailsForDay}
               getUpdatedDraggedEvent={({ event, columnMoves }) => {
-                const newIndex = index + columnMoves;
+                let newIndex = index + columnMoves;
+
+                if (visibleEventGroups) {
+                  let count = columnMoves < 0 ? -1 : 1;
+                  while (
+                    !visibleEventGroups.includes(eventGroups[newIndex].id)
+                  ) {
+                    newIndex = index + columnMoves + count;
+                    // We want to continue moving in the eventGroups array
+                    // until we find the correct index. We have to move in
+                    // the correct direction depending on if columnMoves is
+                    // positive or negative
+                    count = columnMoves < 0 ? count - 1 : count + 1;
+                  }
+                }
+
```

Comment:
This would run in an infinite loop if `visibleEventGroups = []`, right?

Reply:
fixed!